app/services/slack_service.rb

```diff
@@ -0,0 +1,183 @@
+module SlackService
+  class InvalidRequest < RuntimeError;
+  end
+  class InvalidCommand < RuntimeError;
+  end
+
+  public
+
+  def create_post(command_text, team_id, user_id)
+    slack_users = command_text.scan(/(?=<).*?(?<=>)/)
+
+    receivers = []
+
+    slack_users.each do |slack_user|
+      id = slack_user[slack_user.index('@') + 1..slack_user.index('|') - 1]
+      user = User.where(slack_id: id).take
+
+      if user == nil
+        name = slack_user[slack_user.index('|') + 1..slack_user.index('>') - 1]
```

Comment:
Maybe you could extract this (and `id = ...` above) into some helper functions. There's a lot of detail in there that's not really relevant if you're just trying to read what this function does.